### PR TITLE
change internal looping in NuRadioMC

### PR DESCRIPTION
**Pull request by cg-laser**
corresponding PR to https://github.com/nu-radio/NuRadioMC/pull/208

### CODE REVIEW

**Comment by christophwelling** [comment]
Looks good so far. Can you add a has_shower method to the event class to check if there is already a shower with a given ID?

File: NuRadioReco/framework/sim_channel.py
Comment (cg-laser) [reply NuRadioReco/framework/sim_channel.py:12]: yes that would be smarter. I'll change it

File: NuRadioReco/framework/sim_station.py
Comment (cg-laser) [reply NuRadioReco/framework/sim_station.py:38]: I don't understand this comment.

File: NuRadioReco/modules/channelAddCableDelay.py
Comment (cg-laser) [reply NuRadioReco/modules/channelAddCableDelay.py:6]: There is a new model that calculates the efield per channel (to get the "simChannel" voltage traces). This module does not include the cable delays. I found it more logic to have a separate module for it. 
In case of the efieldToVoltageConverter this is not possible because we don't have the intermediate step of calculating simchannel traces.

File: NuRadioReco/modules/channelGenericNoiseAdder.py
Comment (cg-laser) [reply NuRadioReco/modules/channelGenericNoiseAdder.py:189]: jep, will change it

**Comment by anelles** [comment]
@cg-laser per code cleaning up, we should try to merge this. @christophwelling you would need to accepted the changes in order to make is mergable.

File: NuRadioReco/framework/sim_station.py
Comment (christophwelling) [reply NuRadioReco/framework/sim_station.py:38]: Both the Station and the SimStation class now have a iter_channels method that does the same. So it would be better to define this once in the BaseStation class

File: NuRadioReco/modules/channelAddCableDelay.py
Comment (christophwelling) [reply NuRadioReco/modules/channelAddCableDelay.py:6]: Okay, makes sense

**Review by christophwelling — APPROVED** [review]

**Comment by christophwelling** [comment]
Please put the iter_channels method from the Station and SimStation class into the BaseStation class, then I think this can be merged.

File: NuRadioReco/framework/sim_station.py
Comment (cg-laser) [reply NuRadioReco/framework/sim_station.py:38]: they are not exactly the same. The identifiers for the channels are different. We could move the data structure into the basestation and also the iter_channels function. Also the object itself is different. One is a SimChannel and the other one is a Channel. 
So I fear it might get more confusing than helping to move part of the code to the base station.

Comment (christophwelling) [reply NuRadioReco/framework/sim_station.py:38]: Hmm, I see. Hard to say which solution is better. So I'm fine with either. Have you tested if Channel and SimChannel are similar enough that you can use modules like the filter and resampler on both (by just passing the SimStation instead of the Station)? I think that would be very useful

Comment (cg-laser) [reply NuRadioReco/framework/sim_station.py:38]: SimChannel inherits from Channel. The only difference is that it has two more identifiers, the ray tracing and shower id. So all modules that run on the channel should also run on the simchannel. I'm actually using it in NuRadioMC .

**Review by christophwelling — APPROVED** [review]

File: NuRadioReco/framework/sim_station.py
Comment (christophwelling) [reply NuRadioReco/framework/sim_station.py:38]: Great, then I don't have anything to complain about any more ;-)

**Comment by cg-laser** [comment]
The SouthPole test will fail because of the random seeds changes. So I need to update this. But before that, I need to merge the corresponding NuRadioMC changes into the internal-looping branch. For this I'm waiting on @anelles feedback/approval on the NuRadioMC pull request (the noise temperature one).

**Pull request merged**
